src/resources/resources.service.ts

```diff
@@ -19,17 +19,20 @@ export class ResourcesService {
     ) {
         const { url, title, userId } = createResourceDto;
 
+        // make sure teamId exists
+        await this.checkTeamExists(teamId);
+
         // check if this team has already added this resource's URL
         const teamMember = await this.prisma.voyageTeamMember.findFirst({
```

Comment:
Can you throw an error in updateResource too - similar to what you are doing in removeResource?

Reply:
Can you remove userId from the delete body now that we have auth working and the userId is in the request header? Example:
    /```
/Can only delete if loggedIn userId mataches addedBy userId
    @UseGuards(JwtAuthGuard)
    @ApiBearerAuth()
    @Delete("/features/:featureId")
    async deleteFeature(
        @Request() req,
        @Param("featureId", ParseIntPipe) featureId: number,
    ) {
        const feature = await this.featuresService.findOneFeature(featureId);

        if (!feature) {
            throw new NotFoundException(
                `featureId (id: ${featureId}) does not exist.`,
            );
        }

        if (feature.addedBy.member.id === req.user.userId) {
            const deletedFeature =
                await this.featuresService.deleteFeature(featureId);
            return deletedFeature;
        } else {
            throw new HttpException("Forbidden", HttpStatus.FORBIDDEN);
        }
    }
```

Reply:
Yes, and I went ahead and replaced the other instances of taking the user ID from the request body with req.user.userId too. Also secured the GET route so that only team members can access it.